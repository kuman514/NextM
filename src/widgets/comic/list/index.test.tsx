import { render } from '@testing-library/react';

import ComicList from '.';

jest.mock(
  'next/link',
  () =>
    function MockedLink(props: React.AnchorHTMLAttributes<HTMLAnchorElement>) {
      return <a {...props} />;
    }
);

describe('ComicList', () => {
  it('should render as expected', () => {
    const { container } = render(
      <ComicList
        label="Avengers"
        comicDataList={[
          {
            id: 1,
            digitalId: 11,
            title: 'Avengers',
            issueNumber: 0,
            variantDescription: '',
            description: 'Avengers assemble',
            modified: new Date(),
            isbn: '',
            upc: '',
            diamondCode: '',
            ean: '',
            issn: '',
            format: '',
            pageCount: 0,
            textObjects: [],
            resourceURI: '',
            urls: [],
            thumbnail: {
              path: 'https://avenge.rs/images/avengers',
              extension: 'jpg',
            },
            images: [],
            series: {
              resourceURI: '',
              name: '',
            },
            characters: {
              available: 0,
              returned: 0,
              collectionURI: '',
              items: [],
            },
            stories: {
              available: 0,
              returned: 0,
              collectionURI: '',
              items: [],
            },
          },
          {
            id: 2,
            digitalId: 12,
            title: 'Dr. Strange',
            issueNumber: 0,
            variantDescription: '',
            // prettier-ignore
            description: 'Dormammu, I\'ve come to bargain!',
            modified: new Date(),
            isbn: '',
            upc: '',
            diamondCode: '',
            ean: '',
            issn: '',
            format: '',
            pageCount: 0,
            textObjects: [],
            resourceURI: '',
            urls: [],
            thumbnail: {
              path: 'https://avenge.rs/images/dr_strange',
              extension: 'jpg',
            },
            images: [],
            series: {
              resourceURI: '',
              name: '',
            },
            characters: {
              available: 0,
              returned: 0,
              collectionURI: '',
              items: [],
            },
            stories: {
              available: 0,
              returned: 0,
              collectionURI: '',
              items: [],
            },
          },
          {
            id: 3,
            digitalId: 13,
            title: 'Avergers: Endgame',
            issueNumber: 0,
            variantDescription: '',
            // prettier-ignore
            description: 'We\'re in the endgame now.',
            modified: new Date(),
            isbn: '',
            upc: '',
            diamondCode: '',
            ean: '',
            issn: '',
            format: '',
            pageCount: 0,
            textObjects: [],
            resourceURI: '',
            urls: [],
            thumbnail: {
              path: 'https://avenge.rs/images/avengers_endgame',
              extension: 'jpg',
            },
            images: [],
            series: {
              resourceURI: '',
              name: '',
            },
            characters: {
              available: 0,
              returned: 0,
              collectionURI: '',
              items: [],
            },
            stories: {
              available: 0,
              returned: 0,
              collectionURI: '',
              items: [],
            },
          },
        ]}
      />
    );
    expect(container.firstChild).toMatchSnapshot();
  });
});
