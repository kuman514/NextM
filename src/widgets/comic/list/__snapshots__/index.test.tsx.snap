// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ComicList should render as expected 1`] = `
<div
  class="w-full flex flex-col items-start"
>
  <h1
    class="mt-9 mb-6 text-4xl font-bold undefined"
  >
    Avengers
  </h1>
  <ul
    class="w-full flex flex-row overflow-x-auto gap-4"
  >
    <li>
      <a
        aria-label="list-item-link"
        class="flex flex-col w-48 h-72 justify-center rounded-xl overflow-hidden bg-black hover:bg-red-700 transition"
        href="/comic/1"
      >
        <div
          class="w-48 h-48"
        >
          <img
            alt="Avengers"
            class="w-full h-full object-cover"
            data-nimg="1"
            decoding="async"
            height="192"
            loading="lazy"
            src="/_next/image?url=https%3A%2F%2Favenge.rs%2Fimages%2Favengers.jpg&w=384&q=75"
            srcset="/_next/image?url=https%3A%2F%2Favenge.rs%2Fimages%2Favengers.jpg&w=256&q=75 1x, /_next/image?url=https%3A%2F%2Favenge.rs%2Fimages%2Favengers.jpg&w=384&q=75 2x"
            style="color: transparent;"
            width="192"
          />
        </div>
        <div
          class="flex flex-col justify-center items-center w-full h-36"
        >
          <span
            class="text-white text-center"
          >
            Avengers
          </span>
          <span
            class="text-white text-center"
          />
        </div>
      </a>
    </li>
    <li>
      <a
        aria-label="list-item-link"
        class="flex flex-col w-48 h-72 justify-center rounded-xl overflow-hidden bg-black hover:bg-red-700 transition"
        href="/comic/2"
      >
        <div
          class="w-48 h-48"
        >
          <img
            alt="Dr. Strange"
            class="w-full h-full object-cover"
            data-nimg="1"
            decoding="async"
            height="192"
            loading="lazy"
            src="/_next/image?url=https%3A%2F%2Favenge.rs%2Fimages%2Fdr_strange.jpg&w=384&q=75"
            srcset="/_next/image?url=https%3A%2F%2Favenge.rs%2Fimages%2Fdr_strange.jpg&w=256&q=75 1x, /_next/image?url=https%3A%2F%2Favenge.rs%2Fimages%2Fdr_strange.jpg&w=384&q=75 2x"
            style="color: transparent;"
            width="192"
          />
        </div>
        <div
          class="flex flex-col justify-center items-center w-full h-36"
        >
          <span
            class="text-white text-center"
          >
            Dr. Strange
          </span>
          <span
            class="text-white text-center"
          />
        </div>
      </a>
    </li>
    <li>
      <a
        aria-label="list-item-link"
        class="flex flex-col w-48 h-72 justify-center rounded-xl overflow-hidden bg-black hover:bg-red-700 transition"
        href="/comic/3"
      >
        <div
          class="w-48 h-48"
        >
          <img
            alt="Avergers: Endgame"
            class="w-full h-full object-cover"
            data-nimg="1"
            decoding="async"
            height="192"
            loading="lazy"
            src="/_next/image?url=https%3A%2F%2Favenge.rs%2Fimages%2Favengers_endgame.jpg&w=384&q=75"
            srcset="/_next/image?url=https%3A%2F%2Favenge.rs%2Fimages%2Favengers_endgame.jpg&w=256&q=75 1x, /_next/image?url=https%3A%2F%2Favenge.rs%2Fimages%2Favengers_endgame.jpg&w=384&q=75 2x"
            style="color: transparent;"
            width="192"
          />
        </div>
        <div
          class="flex flex-col justify-center items-center w-full h-36"
        >
          <span
            class="text-white text-center"
          >
            Avergers: Endgame
          </span>
          <span
            class="text-white text-center"
          />
        </div>
      </a>
    </li>
  </ul>
</div>
`;
